File: golang-backend/internal/handlers/projecttypehandler/dto.go
package projecttypehandler

import "github.com/aportela/doneo/internal/handlers"

type addRequest struct {
	Name     string `json:"name"`
	HexColor string `json:"hexColor"`
}

type updateRequest struct {
	Name     string `json:"name"`
	HexColor string `json:"hexColor"`
}

type filterRequest struct {
	Name *string `json:"name"`
}

type searchRequest struct {
	Pager  handlers.PagerRequest `json:"pager"`
	Order  handlers.OrderRequest `json:"order"`
	Filter *filterRequest        `json:"filter"`
}

type projectTypeResponse struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	HexColor string `json:"hexColor"`
}

type searchResponse struct {
	ProjectTypes []projectTypeResponse  `json:"projectTypes"`
	Pager        handlers.PagerResponse `json:"pager"`
}

File: vite-vue-ts-frontend/src/modules/project-priorities/types/dto.ts
import type {
  PagerRequest,
  PagerResponse,
} from "../../../shared/types/dto/pager";
import type { Order } from "../../../shared/types/dto/order";

export type AddRequest = {
  name: string;
  HexColor: string;
};

export type UpdateRequest = {
  id: string;
  name: string;
  HexColor: string;
};

type SearchRequestFilter = {
  name?: string;
};

export type SearchRequest = {
  pager: PagerRequest;
  order: Order;
  filter?: SearchRequestFilter;
};

export type ProjectPriorityResponse = {
  id: string;
  name: string;
  hexColor: string;
  index: number;
};

export type SearchResponse = {
  projectPriorities: ProjectPriorityResponse[];
  pager: PagerResponse;
};
